data-pipeline/src/trace_exporter.rs

```diff
@@ -17,13 +20,17 @@ use dogstatsd_client::{new_flusher, Client, DogStatsDAction};
 use either::Either;
 use hyper::http::uri::PathAndQuery;
 use hyper::{Body, Method, Uri};
-use log::error;
+use log::{error, info};
 use std::sync::{Arc, Mutex};
 use std::time::Duration;
 use std::{borrow::Borrow, collections::HashMap, str::FromStr, time};
 use tokio::{runtime::Runtime, task::JoinHandle};
 use tokio_util::sync::CancellationToken;
 
+const DEFAULT_STATS_ELIGIBLE_SPAN_KINDS: [&str; 4] = ["client", "server", "producer", "consumer"];
+const STATS_ENDPOINT: &str = "/v0.6/stats";
+const INFO_ENDPOINT: &str = "/info";
```

Comment:
I wonder if we're not properly encapsulating the stats computation logic? It's not obvious to me why all of these changes should live in the trace exporter. These consts , the `StatsComputationStatus` enum, and the stats specific start/shutdown logic are all examples.

Reply:
I agree, yet this would probably require a larger refactor. I'll try to improve it without refactoring too much.